File: src/screen-size.js
class ScreenSize {
	constructor(width, height) {
		this._width = width;
		this._height = height;
	}

	get width() {
		return this._width;
	}

	get height() {
		return this._height;
	}

	fromJS(output) {
		output(this._width, this._height);
	}

	toJSON() {
		const {width, height} = this;
		return {width, height};
	}

	toString(output = this) {
		return `${output.width}x${output.height}`;
	}
}

module.exports = ScreenSize;

File: src/screen-size.cc
#include "screen-size.h"

#include "nbind/nbind.h"

ScreenSize::ScreenSize(size_t width, size_t height) {
	w = width;
	h = height;
}

size_t ScreenSize::getWidth() {
	return w;
}

size_t ScreenSize::getHeight() {
	return h;
}

void ScreenSize::toJS(nbind::cbOutput output) {
	output(w, h);
}



NBIND_CLASS(ScreenSize) {
  construct<size_t, size_t>();
  method(getWidth);
  method(getHeight);
}


